pandas/_libs/writers.pyx

```diff
@@ -144,7 +144,7 @@ cpdef inline Py_ssize_t word_len(object val):
         Py_ssize_t l = 0
 
     if isinstance(val, str):
-        l = PyUnicode_GET_SIZE(val)
+        l = PyUnicode_GET_LENGTH(val)
```

Comment:
this can be done independently of the rest of the PR?

Reply:
Yeah, do you want to split it out?